ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/txselection/BlockTransactionSelector.java

```diff
@@ -339,24 +340,27 @@ public TransactionSelectionResult evaluatePendingTransaction(
   @Override
   public boolean commit() {
     // only add this tx to the selected set if it is not too late,
-    // this need to be done synchronously to avoid that a concurrent timeout
+    // this needs to be done synchronously to avoid that a concurrent timeout
     // could start packing a block while we are updating the state here
+    final boolean isTooLate;
     synchronized (isTimeout) {
-      if (!isTimeout.get()) {
+      isTooLate = isTimeout.get();
+      if (!isTooLate) {
         selectorsStateManager.commit();
         txWorldStateUpdater.commit();
```

Comment:
do we still need this commit ? I think this one is done in the processTransaction ?

Reply:
ah ok I think I see why it's needed.